### PR TITLE
Transport parameters

### PR DESCRIPTION
**Pull request by lalmei**
## Description

waiting for confirmation, or test on minio setup 

### General Checklist

* [x] Tests added for this feature/bug
      if it was a bug, test must cover it.
* [x] Conform by the style guides, by using formatter
* [x] Documentation updated
* [x] (optional) Please add a label to your PR

### CODE REVIEW

File: tests/unit/app/test_writers.py [hunk 1]
```diff
@@ -71,11 +79,22 @@ def test_s3_writer(df_lending_club, moto_boto, s3_all_config_path):
 
     client = boto3.client("s3")
     objects = client.list_objects(Bucket="mocked_bucket")
-
     for idx, each_objc in enumerate(objects["Contents"]):
         assert each_objc["Key"] == object_keys[idx]
 
 
+def test_s3_writer_transport_config(df_lending_club, s3_transport_config_path):
```
Comment (jamie256) [inline tests/unit/app/test_writers.py]: it doesn't look like you are using the df_lending_club here, can you remove it?

File: tests/unit/app/test_session_config.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+def test_config():
+    name = uuid4()
+    pipeline = uuid4()
+    session_config = SessionConfig(name, pipeline, writers=[])
+
+    assert session_config.project == name
+    assert session_config.pipeline == pipeline
+
+    assert session_config.writers == []
+    # session = session_from_config(session_config)
```
Comment (jamie256) [inline tests/unit/app/test_session_config.py]: remove?

File: examples/.gitignore [hunk 3]
```diff
@@ -1,4 +1,5 @@
 whylogs-output/
+whylogs-image/
```
Comment (jamie256) [inline examples/.gitignore:2]: Is this required as part of the transport_parameters changes?

**Review by jamie256 — COMMENTED** [review]
Left some minor comments and questions but looks great to avoid the default session in boto3 client!

File: tests/unit/app/test_writers.py (hunk 1 above)
Comment (lalmei) [reply tests/unit/app/test_writers.py]:
```suggestion
def test_s3_writer_transport_config(s3_transport_config_path):
```

File: tests/unit/app/test_session_config.py (hunk 2 above)
Comment (lalmei) [reply tests/unit/app/test_session_config.py]:
```suggestion
```

File: examples/.gitignore (hunk 3 above)
Comment (lalmei) [reply examples/.gitignore:2]: i dont think this change requires a separate issue/PR and comes up during the testing of the notebooks locally rather annoying.

Comment (jamie256) [reply examples/.gitignore:2]: Sure, no problem. Wanted to check it was intentional.

**Review by jamie256 — APPROVED** [review]
LGTM!

**Pull request merged**